iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -2021,4 +2023,34 @@ public List<Partition> getPartitionsByExpr(org.apache.hadoop.hive.ql.metadata.Ta
       throw new SemanticException(String.format("Error while fetching the partitions due to: %s", e));
     }
   }
+
+  @Override
+  public boolean supportsMergeFiles() {
+    return true;
+  }
+
+  @Override
+  public List<FileStatus> getMergeInputFiles(Properties properties) throws IOException {
+    String tableName = properties.getProperty(Catalogs.NAME);
+    String snapshotRef = properties.getProperty(Catalogs.SNAPSHOT_REF);
+    Configuration configuration = SessionState.getSessionConf();
+    List<JobContext> originalContextList = generateJobContext(configuration, tableName, snapshotRef);
+    List<JobContext> jobContextList = originalContextList.stream()
+            .map(TezUtil::enrichContextWithVertexId)
+            .collect(Collectors.toList());
+    if (originalContextList.isEmpty()) {
+      return Collections.emptyList();
+    }
+    List<FileStatus> dataFiles = Lists.newArrayList();
+    for (DataFile dataFile : new HiveIcebergOutputCommitter().getWrittenFiles(jobContextList)) {
```

Comment:
1. why do you need new instance of HiveIcebergOutputCommitter, can't you make the method static?
2. Why do you need to enrichContextWithVertexId again inside of `getWrittenFiles`?

Reply:
> why do you need new instance of HiveIcebergOutputCommitter, can't you make the method static? Maybe we should move it to Iceberg util class

Kept it in the same place so that we could make use of existing helper functions already present in HiveIcebergOutputCommitter.

> Why do you need to enrichContextWithVertexId again inside of getWrittenFiles?

Removed and doing only once.

>  Why don't we return List<FileStatus> dataFiles right away, but add 1 more foreach with transformation?

Added functionality to directly convert it into `Filestatus` objects.